keps/sig-multicluster/5313-placement-decision-api/README.md

```diff
@@ -0,0 +1,429 @@
+* Support continuous rescheduling: decision list may be updated.
+* Guarantee that every `clusterName` entry matches a `ClusterProfile.metadata.name` in the same inventory.
+* Guarantee that every `clusterName` entry is in the same namespace as `PlacementDecision.metadata.namespace`.
+* Provide label conventions so consumers can retrieve all slices of one placement.
+* Leave room for schedulers implementations.
+
+### Non-Goals
+
+* Describing how a scheduler made its choice (Placement API spec).
+* Describing how consumers access selected clusters.
+* Embedding orchestration logic or consumer feedback in `PlacementDecision`.
+* Replace/overlap Work API which is responsible for actually applying the workload.
+
+## Proposal
+
+### API Definition
+
+```
+// PlacementDecision publishes the selected clusters for one Placement.
+type PlacementDecision struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  // Up to 100 ClusterDecisions per object (slice) to stay well below the etcd limit.
+  Decisions []ClusterDecision `json:"decisions"`
+}
+
+// Label that links all slices to their originating Placement request.
+const PlacementLabel = "multicluster.x-k8s.io/placement"
+
+// Label that link all slices to which scheduler produced the slices.
+const SchedulerLabel = "multicluster.x-k8s.io/scheduler"
+
+type ClusterDecision struct {
+  // MUST match a ClusterProfile.metadata.name in the same namespace.
+  ClusterName string `json:"clusterName"`
+}
+```
+
+### API Example
+
+```
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: PlacementDecision
+metadata:
+  name: app-placement-decision-1
+  namespace: argocd
+  labels:
+	  multicluster.x-k8s.io/placement: app-placement
+	  multicluster.x-k8s.io/scheduler: multicluster-placement-controller
+decisions:
+- clusterName: cluster1
+- clusterName: cluster2
+```
+
+### Terminology
+
+- **Placement**: A scheduler request that asks "where should this workload run?".
+
+- **Scheduler**: Placement controller that writes `PlacementDecisions` based on `ClusterProfiles` and
+  scheduling/placement requirements/specs.
+
+- **Consumer**: Any controller (GitOps engine, workload orchestrator, progressive rollout controller, AI/ML pipeline)
+  that watches `PlacementDecisions` and acts.
+
+### Slicing
+
+* Following [EndpointSlice](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) design,
+  a single Placement can fan out to N `PlacementDecision` slices,
+  each limited to 100 clusters (`EndpointSlice`'s default).
+* All slices for one `Placement` MUST carry the same
+  `multicluster.x-k8s.io/placement=<placement-name>` label so consumers can List with a label selector.
+* The scheduler that owns the slice MUST label it with
+  `multicluster.x-k8s.io/scheduler=<scheduler-name>` for observability.
+
+### Lifecycle
+- **Create**: The scheduler creates the slice with the list of clusters in the decisions,
```

Comment:
we might need some ownership explanation, e.g. scheduler is responsible for the api, consumers should not update but only read.

Reply:
Absolutely! Added a "Ownership" section as suggested.